lib/ansible/modules/remote_management/manageiq/manageiq_config.py

```diff
@@ -0,0 +1,211 @@
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: manageiq_config
+
+short_description: Module for managing ManageIQ configuration.
+version_added: '2.6'
+author: Ian Tewksbury (@itewk)
+description:
+    - "Module for managing ManageIQ or CloudForms Managment Engine (CFME) configuration via the local rails runner on the Appliance."
+
+options:
+    name:
+        description:
+            - Name of the ManageIQ config element to modify.
+        required: True
+    value:
+        description: Dictionary to set as the value for the ManageIQ config option. Any values not given will be left at current value.
+        required: False
+        default: {}
+    vmdb_path:
+        description: Path to the VMDB directory.
+        required: False
+        default: /var/www/miq/vmdb
+'''
+
+EXAMPLES = '''
+# set the smtp settings
+- manageiq_config:
+    name: smtp
+    value:
+      from: cfme@example.com
+      host: postfix.example.com
+      port: 25
+      domain: example.com
+
+# set generic worker memory threshold
+- manageiq_config:
+    name: workers
+    value:
+      worker_base:
+        queue_worker_base:
+          generic_worker:
+            memory_threshold: '900.megabytes'
+'''
+
+RETURN = '''
+name:
+    description: Name of the ManageIQ config option that is being modified.
+    type: str
+    returned: always
+original_value:
+    description: The original value of the ManageIQ config option being modified before modification.
+    type: dict
+    returned: always
+value:
+    description: The value of the ManageIQ config option being modified after modification.
+    type: dict
+    returned: always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import json
+import time
+import collections
+import copy
+
+
+# @source https://gist.github.com/angstwad/bf22d1822c38a92ec0a9
+def dict_merge(dct, merge_dct):
+    """ Recursive dict merge. Inspired by :meth:``dict.update()``, instead of
+    updating only top-level keys, dict_merge recurses down into dicts nested
+    to an arbitrary depth, updating keys. The ``merge_dct`` is merged into
+    ``dct``.
+    :param dct: dict onto which the merge is executed
+    :param merge_dct: dct merged into dct
+    :return: None
+    """
+    for k, v in merge_dct.items():
+        if (k in dct and isinstance(dct[k], dict)
+                and isinstance(merge_dct[k], collections.Mapping)):
+            dict_merge(dct[k], merge_dct[k])
+        else:
+            dct[k] = merge_dct[k]
+
+
+def get_manageiq_config_value(module, name):
+    """ Gets the current value for the given config option.
+    :param module: AnsibleModule making the call
+    :param name: ManageIQ config option name
+    :return: dict of value of the ManageIQ config option
+    """
+    returncode, out, err = module.run_command([
+        "rails",
+        "r",
+        "puts MiqServer.my_server.get_config(:%s).config.to_json" % (name)
```

Comment:
FYI, in case you switch to working on whole config, `get_config.config` gives you the whole tree.  (`get_config(:foo).config` is generally equivalent to `get_config.config[:foo]`.)
Also, VMDB::Config is deprecated.  May be better to use `::Settings.to_hash`.

Reply:
@cben I am not familiar with `::Settings.to_hash` could you point me where I would need to look for an example or replacing current `MiqServer.my_server.get_config` call with your suggestion?

Reply:
+1 for @cben's suggestion here.

I would think `::Settings.to_hash` or `MiqServer.my_server.settings_for_resource` would be preferred here. I'm not sure which is the better way to go .. @Fryguy ?

Reply:
I agree with @cben here.  VMDB::Config is deprecated, and really only was kept around for some UI code that was using it.  If you are basically on the server, it's better to use `::Settings.to_hash` (or more specifically `::Settings.to_hash.to_json`)